### PR TITLE
Added tracks for mobile welcome modal

### PR DESCRIPTION
**Pull request by rjchow**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

76-gh-woocommerce/team-ghidorah

Tracks events for the moile app welcome modal: 

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_magic_prompt_view`   |
| What does the track measure      | View count for WooCommerce Mobile App magic link prompt. Note: this does not fire on the return of the user from wordpress.com account connection |
| When does the track fire      | When a user views magic link prompt |
| Event properties      | `jetpack_state`: Status of Jetpack plugin. Values: `full-connection` , `not-installed` , `installing` , `user-cannot-install` , `not-activated` , `userless-connection` , `not-owner-of-connection` <br><br> `repeat_user`: Whether the user has dismissed the modal before. Values: `true`, `false`

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_magic_prompt_return_from_wp_connection`   |
| What does the track measure      |  Count of users who returned from the jetpack installation flow  |
| When does the track fire      | When a user has completed the jetpack installation flow and returns to wpadmin |
| Event properties      | - |

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_magic_prompt_install_connect_click`   |
| What does the track measure      | Clicks count on Install and Connect button  |
| When does the track fire      | When a user clicks on Install and Connect button |
| Event properties      | - |

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_magic_prompt_send_signin_link_click`   |
| What does the track measure      | Clicks count on Send the sign-in link button  |
| When does the track fire      | When a user clicks on Send the sign-in link button |
| Event properties      | - |

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_magic_prompt_mismatched_wpcom_user_view`   |
| What does the track measure      |  The number of users who have mismatched wpcom accounts  |
| When does the track fire      | When a user clicks on the additional task list item to get the woocommerce mobile app |
| Event properties      | - |

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_magic_prompt_retry_send_signin_link`   |
| What does the track measure      |  The number of users who return to the send link page  |
| When does the track fire      | When a user clicks on the 'send another link' link on the email success page |
| Event properties      | - |

### How to test the changes in this Pull Request:

Same as: https://github.com/woocommerce/woocommerce/pull/34637

To see the tracks events you need to run the command `localStorage.debug="*"` in your browser console.

To see the mismatched connection event you will also need to set up a second admin account to attempt the flow.

A run-through of the install, connect, send magic link sequence should look like this:

image

A run-through of the jetpack-already-installed sequence should look like this:

image

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by ilyasfoo** [comment]
@rjchow Quick one, I noticed in the PR description `wcadmin_magic_prompt_return_from_wp_connection` defined twice, I think that's not intended?

**Comment by rjchow** [comment]
> @rjchow Quick one, I noticed in the PR description `wcadmin_magic_prompt_return_from_wp_connection` defined twice, I think that's not intended?

Thanks @ilyasfoo , copypaste fail!

File: plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/JetpackInstallationStepper.tsx [hunk 1]
```diff
@@ -50,6 +51,9 @@ export const JetpackInstallationStepper = ( {
 								className="install-jetpack-button"
 								onClick={ () => {
 									setIsWaitingForRedirect( true );
+									recordEvent(
+										'wcadmin_magic_prompt_install_connect_click'
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/JetpackInstallationStepper.tsx]:
```suggestion
										'magic_prompt_install_connect_click'
```

The `wcadmin` prefix is not required, otherwise it'll become 
image

**Review by ilyasfoo — CHANGES_REQUESTED** [review]

File: plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/JetpackInstallationStepper.tsx (hunk 1 above)
Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/JetpackInstallationStepper.tsx]: @rjchow Should apply to all `wcadmin_` prefix in this PR

Comment (rjchow) [reply plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/JetpackInstallationStepper.tsx]: how'd I miss that... 🤦 Thanks!

File: plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/SendMagicLinkButton.tsx [hunk 2]
```diff
@@ -9,7 +9,12 @@ export const SendMagicLinkButton = ( {
 }: {
 	onClickHandler: () => void;
 } ) => (
-	<Button className="send-magic-link-button" onClick={ onClickHandler }>
+	<Button
+		className="send-magic-link-button"
+		onClick={ () => {
+			onClickHandler();
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/SendMagicLinkButton.tsx]: I'm curious why this change is required?

**Review by ilyasfoo — DISMISSED** [review]
Something I noticed is when a user clicks on `send another link`, the `wcadmin_magic_prompt_view` track is triggered again. I think this is incorrect behaviour since re-sending shouldn't signal another view track. What do you think?

Since `wcadmin_magic_prompt_return_from_wp_connection` track replaces `wcadmin_magic_prompt_view` when user has returned from WPCOM, do you think it makes sense to add `jetpack_state`? I assume it would hold `full-connection` most of the time, but maybe it could help us debug connection issues in the future.

I have some questions but since the tracks are working as expected, pre-approving 👍

File: plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/SendMagicLinkButton.tsx (hunk 2 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/SendMagicLinkButton.tsx]: argh I didn't revert correctly - I mistakenly put the install-and-connect tracks in here previously and noticed it at some point, so I removed the recordEvent only

**Comment by rjchow** [comment]
> Something I noticed is when a user clicks on send another link, the wcadmin_magic_prompt_view track is triggered again. I think > this is incorrect behaviour since re-sending shouldn't signal another view track. What do you think?

Makes sense, I'll fix that!

> Since wcadmin_magic_prompt_return_from_wp_connection track replaces wcadmin_magic_prompt_view when user has returned > from WPCOM, do you think it makes sense to add jetpack_state? I assume it would hold full-connection most of the time, but 
> maybe it could help us debug connection issues in the future.

it's not possible to get to that screen with anything other than a full-connection because of the condition in the wrapper:

```
			( state === JetpackPluginStates.FULL_CONNECTION &&
				isReturningFromWordpressConnection )
```

**Comment by rjchow** [comment]
@ilyasfoo done! A retry flow should look like this now:
image

I'll add the new event to the issue and PR description

**Review by ilyasfoo — APPROVED** [review]
Thanks for fixing! Tested well, LGTM!
